### PR TITLE
Add findbugs-slf4j as mandatory plugin

### LINKED ISSUE #26: FindBugs slf4j bug pattern
IIRC I comment this already on the ESH repo some time ago before the MOJO moved.
Shouldn't the slf4j bug pattern for FindBugs be a mandatory check?
Currently it seems we need to add the pattern in the plugin configuration:
https://github.com/openhab/static-code-analysis/blame/dd38ca38b698189a7e118e6c0adca894b00cedf4/README.md#L41

I assume this should be configured by the plugin itself similar to the FindBugs, Checkstyle, PMD, ... dependencies, too.

### PR DESCRIPTION
**Pull request by svilenvul**
Closes #26

Signed-off-by: Svilen Valkanov <svilen.valkanov@musala.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,7 +81,7 @@ Parameters:
 | ------ | ------| -------- |
 | **pmdRuleset** | String | Relative path of the XML configuration to use. If not set the default ruleset file will be used |
 | **maven.pmd.version** | String | The version of the maven-pmd-plugin that will be used (Default value is **3.7**)|
-| **pmdPlugins** | Dependency [] | A list with artifacts that contain additional checks for PMD |
+| **pmdPlugins** | ArrayList<Dependency> | A list with artifacts that contain additional checks for PMD |
```
Comment (martinvw) [inline README.md]: List<Dependency> maybe

File: src/main/java/org/openhab/tools/analysis/tools/AbstractChecker.java [hunk 2]
```diff
@@ -154,11 +154,12 @@ protected void executeCheck(String groupId, String artifactId, String version, S
...
      * the analysis tools.
      *
-     * @param additionalArtifacts - artifacts that contain custom checks
      * @param coreDependency - a dependency to the tool itself, used to specify a certain version of the used tool
+     * @param additionalArtifacts - artifacts that contain custom checks
+     *
      * @return dependencies
      */
-    protected Dependency[] getDependencies(Dependency[] additionalArtifacts, Dependency coreDependency) {
+    protected Dependency[] getDependencies(Dependency coreDependency, ArrayList<Dependency> additionalArtifacts) {
```
Comment (martinvw) [inline src/main/java/org/openhab/tools/analysis/tools/AbstractChecker.java]: List

File: src/main/java/org/openhab/tools/analysis/tools/CheckstyleChecker.java [hunk 3]
```diff
@@ -55,7 +56,7 @@
      * A list with artifacts that contain additional checks for Checkstyle
      */
     @Parameter
-    private Dependency[] checkstylePlugins;
+    private ArrayList<Dependency> checkstylePlugins;
```
Comment (martinvw) [inline src/main/java/org/openhab/tools/analysis/tools/CheckstyleChecker.java]: List

File: src/main/java/org/openhab/tools/analysis/tools/FindBugsChecker.java [hunk 4]
```diff
@@ -74,10 +75,16 @@
...
+     */
+    @Parameter(property = "findbugs.slf4j.version", defaultValue = "1.2.4")
+    private String findBugsSlf4jPluginVersion;
+
+    /**
      * A list with artifacts that contain additional checks for FindBugs
      */
     @Parameter
-    private Dependency[] findbugsPlugins;
+    private ArrayList<Dependency> findbugsPlugins;
```
Comment (martinvw) [inline src/main/java/org/openhab/tools/analysis/tools/FindBugsChecker.java]: List

File: src/main/java/org/openhab/tools/analysis/tools/PmdChecker.java [hunk 5]
```diff
@@ -51,7 +52,7 @@
      * A list with artifacts that contain additional checks for PMD
      */
     @Parameter
-    private Dependency[] pmdPlugins;
+    private ArrayList<Dependency> pmdPlugins;
```
Comment (martinvw) [inline src/main/java/org/openhab/tools/analysis/tools/PmdChecker.java]: List

**Comment by svilenvul** [comment]
Thanks again @martinvw  for looking at the code :-) !

**Pull request merged**